pennylane_lightning_gpu/src/simulator/interfaces/torch/Bindings.cpp

```diff
@@ -0,0 +1,343 @@
+// Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <set>
+#include <tuple>
+#include <type_traits>
+#include <variant>
+#include <vector>
+
+#include "SVCudaTorch.hpp"
+#include <torch/extension.h>
+
+#include "pybind11/complex.h"
+#include "pybind11/numpy.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+
+/// @cond DEV
+namespace {
+using namespace Pennylane::CUDA;
+using namespace Pennylane::CUDA::Util;
+using std::complex;
+using std::set;
+using std::string;
+using std::vector;
+
+namespace py = pybind11;
+
+/**
+ * @brief Templated class to build all required precisions for Python module.
+ *
+ * @tparam SVtype Templated state-vector class type.
+ * @tparam PrecisionT Precision of the statevector data.
+ * @tparam ParamT Precision of the parameter data.
+ * @param m Pybind11 module.
+ */
+template <template <typename...> class SVType, class PrecisionT, class ParamT>
+void StateVectorCuda_class_bindings(py::module &m) {
+    using np_arr_c = py::array_t<std::complex<ParamT>,
+                                 py::array::c_style | py::array::forcecast>;
+
+    /// Enable module name to be based on size of complex datatype
+    /// Module naming convention:
+    /// LightningGPU_<name of C++ class>_C<num bits in complex value>
+    const std::string bitsize =
+        std::to_string(sizeof(std::complex<PrecisionT>) * 8);
+    std::string cls_name =
+        "LightningGPU_" + SVType<PrecisionT>::getClassName() + "_C" + bitsize;
+
+    py::class_<SVType<PrecisionT>>(m, cls_name.c_str())
```

Comment:
I am also slightly worried about the code publication here (with `bindings/Binding.cpp`).

Reply:
I agree that it is better to separate this module, but can we just make move this function to another header file and use them in both cases?